celery/app/utils.py

```diff
@@ -190,6 +195,20 @@ def humanize(self, with_defaults=False, censored=True):
             f'{key}: {pretty(value, width=50)}'
             for key, value in self.table(with_defaults, censored).items())
 
+    def maybe_warn_deprecated_settings(self):
+        # TODO: Remove this method in Celery 6.0
+        if self.deprecated_settings:
+            from celery.utils import deprecated
+            from celery.app.defaults import _TO_NEW_KEY
+            for setting in self.deprecated_settings:
+                deprecated.warn(description=f'The {setting!r} setting',
+                                removal='6.0.0',
+                                alternative=f'Use the {_TO_NEW_KEY[setting]} instead')
```

Comment:
This is going to be pretty noisy (people who haven't changed one setting probably have many that aren't changed), but I don't see a better way to do it.